src/store/actions/api.actions.js

```diff
@@ -57,7 +67,30 @@ export const willClone = data => ({
   payload: data,
 });
 
-export const fetchAPI = (pathname) => dispatch => {
+export const deleteAPI = (apiName, callback) => dispatch => {
+  dispatch(deleteAPIRequest());
+
+  return client.delete(`apis/${apiName}`)
+    .then(response => {
+      dispatch(deleteAPISuccess());
+      dispatch(openResponseModal({
```

Comment:
Success/failure feedback should be separated onto two parts as we discussed previously. Should we implement in this PR or I believe the separate one is preferable? cc @nik-kor